src/librustc_middle/query/mod.rs

```diff
@@ -878,7 +886,7 @@ rustc_queries! {
 
         /// Identifies the entry-point (e.g., the `main` function) for a given
         /// crate, returning `None` if there is no entry point (such as for library crates).
-        query entry_fn(_: CrateNum) -> Option<(DefId, EntryFnType)> {
+        query entry_fn(_: CrateNum) -> Option<(LocalDefId, EntryFnType)> {
```

Comment:
I am confused by this. I thought `LocalDefId` means "in the local crate", but `entry_fn` takes a `CrateNum` parameter and thus might return something in a different crate?

Reply:
Since the query is cached, if it is called from another crate, it'll just fetch the cached version. This kind of trickery is used a lot

Reply:
Oh, nevermind, that's wrong apparently, at least I'm reading @eddyb's comment below to meaning these are actually no-argument crates